backend/src/main/java/de/bund/digitalservice/ris/norms/application/service/ModificationValidator.java

```diff
@@ -0,0 +1,525 @@
+            () ->
+                new XmlContentException(
+                    "For norm with Eli (%s): mod href is empty in article with eId %s"
+                        .formatted(eli, articleEId),
+                    null));
+  }
+
+  private TextualMod getTextualModMatching(
+      List<TextualMod> passiveMods, String articleModTargetHrefEId) {
+    List<TextualMod> passiveModsMatching =
+        passiveMods.stream()
+            .filter(
+                pm -> {
+                  Href destinationHref =
+                      pm.getDestinationHref()
+                          .orElseThrow(() -> new XmlContentException("TBD 8", null));
+                  String destinationHrefEId =
+                      destinationHref
+                          .getEId()
+                          .orElseThrow(() -> new XmlContentException("TBD 9", null));
+                  return Objects.equals(destinationHrefEId, articleModTargetHrefEId);
+                })
+            .toList();
+
+    if (passiveModsMatching.size() > 1) throw new XmlContentException("TBD 10", null);
+    if (passiveModsMatching.isEmpty()) throw new XmlContentException("TBD 11", null);
+    return passiveModsMatching.getFirst();
+  }
+
+  private String getTextualModEId(String eli, TextualMod tm) {
+    return tm.getEid()
+        .orElseThrow(
+            () -> // TODO repair test
+            new XmlContentException(
+                    "For norm with Eli (%s): TextualMod eId empty.".formatted(eli), null));
+  }
+
+  private Href getTextualModSourceHref(String eli, TextualMod tm, String textualModEId) {
+    // TODO test for that throw
+    return tm.getSourceHref()
+        .orElseThrow(
+            () ->
+                new XmlContentException(
+                    "For norm with Eli (%s): PassiveModification TextualMod Source Href is empty where textualMod eId is %s"
+                        .formatted(eli, textualModEId),
+                    null));
+  }
+
+  private Href getTextualModDestinationHref(String eli, TextualMod tm, String textualModEId) {
+    // TODO test for that throw
+    return tm.getDestinationHref()
+        .orElseThrow(
+            () ->
+                new XmlContentException(
+                    "For norm with Eli (%s): PassiveModification TextualMod Destination Href is empty where textualMod eId is %s"
+                        .formatted(eli, textualModEId),
+                    null));
+  }
+
+  private CharacterRange getHrefCharacterRange(String eli, Href h, String message) {
+    return h.getCharacterRange()
+        .orElseThrow(
+            () ->
+                new XmlContentException(
+                    "For norm with Eli (%s): %s".formatted(eli, message), null));
+  }
+
+  private String getHrefEId(String eli, Href h, String message) {
+    return h.getEId()
+        .orElseThrow(
+            () ->
+                new XmlContentException(
+                    "For norm with Eli (%s): %s".formatted(eli, message), null));
```

Comment:
for the getters that build together an error message like this (so with a message parameter) I'm questioning if we really need this additional layer of indirection. Especially as they (`getHrefEId`, `getHrefCharacterRange`,  and `getModOldText`) are only used once each. I think just doing this in the place they are called directly would simplify the logic a bit

Reply:
Resoved [here](https://github.com/digitalservicebund/ris-norms/commit/1d77482b0a1443c72d372a7d9840ef52fb657f61)